x-pack/plugins/security_solution/public/common/components/event_details/cti_details/threat_details_view.tsx

```diff
@@ -138,23 +163,82 @@ const EnrichmentAccordion: React.FC<{
   );
 };
 
-const EnrichmentSection: React.FC<{ enrichments: CtiEnrichment[] }> = ({ enrichments }) => (
-  <>
-    {enrichments
-      .sort((a, b) => getFirstSeen(b) - getFirstSeen(a))
-      .map((enrichment, index) => (
-        <EnrichmentAccordion
-          enrichment={enrichment}
-          index={index}
-          enrichmentsLength={enrichments.length}
-        />
-      ))}
-  </>
-);
+const getMessagesFromType = (type?: EnrichmentType) => {
```

Comment:
It seems like this function is doing two things (if we exclude `dataTestSubj`, see below comment about that):

1. Determining the title for a given enrichment type
2. Determining content for the "no data" screen of a given enrichment type

To me, those responsibilities sound like candidates for two independent components, e.g. `EnrichmentTitle` and `NoEnrichments`. `ThreatDetailsView` is now doing a lot of work, and I think we could do away with this function and clean things up along responsibility lines by refactoring those responsibilities into new components. The old `NoEnrichmentsPanel` was too complicated for this PR, so it was understandably deleted, but I think the separation that something like that provides really helps with maintenance in the long term.

As with anything, single responsibility is an ideal and not always feasible/useful, but I think that in general we should strive for smaller, isolated, recomposable components over inextricable ones like this, even if we don't end up ... extricating 😉

Reply:
this function is removed